lib/commons/aria/get-role-segments.js

```diff
@@ -0,0 +1,44 @@
+/* global aria, axe */
+
+/**
+ * Returns all roles applicable to element in a list
+ *
+ * @method getRoleSegments
+ * @memberof axe.commons.aria
+ * @instance
+ * @param {Element} node
+ * @returns {Array} Roles list or empty list
+ */
+
+aria.getRoleSegments = function getRoleSegments(node) {
```

Comment:
Either add tests for this, or don't move this into a commons. My preference would be the latter. I think you've made this PR much larger than it needed to be for the fix.

Reply:
There are tests (https://github.com/dequelabs/axe-core/pull/1118/files#diff-c01d2c9028bf4f6a16bace64d8ce59daR1) indeed . They were at the bottom of the PR.

I need to use `getRoleSegments` in 2 places, one within the `matches`, the other within the `check`, hence the abstraction to commons.

But believe I could use `getRole` in `matches` and I will move `getRoleSegments` out of commons.